src/test/java/org/terasology/launcher/game/TestGameRunner.java

```diff
@@ -15,157 +15,172 @@
-import org.slf4j.Logger;
-import org.slf4j.LoggerFactory;
+import org.hamcrest.Matchers;
+import org.junit.Before;
+import org.junit.Test; // junit because Spf4jTestLogRunner is not a Jupiter Extension yet
+import org.junit.runner.RunWith;
+import org.spf4j.log.Level;
+import org.spf4j.test.log.LogAssert;
+import org.spf4j.test.log.TestLoggers;
+import org.spf4j.test.log.annotations.CollectLogs;
+import org.spf4j.test.log.annotations.ExpectLog;
+import org.spf4j.test.log.annotations.PrintLogs;
+import org.spf4j.test.log.junit4.Spf4jTestLogJUnitRunner;
+import org.spf4j.test.matchers.LogMatchers;
 
 import java.io.ByteArrayInputStream;
 import java.io.IOException;
 import java.io.InputStream;
 
+import static org.hamcrest.Matchers.allOf;
+import static org.hamcrest.Matchers.containsString;
 import static org.mockito.ArgumentMatchers.any;
 import static org.mockito.Mockito.anyInt;
-import static org.mockito.Mockito.atLeastOnce;
-import static org.mockito.Mockito.eq;
 import static org.mockito.Mockito.mock;
-import static org.mockito.Mockito.verify;
+import static org.mockito.Mockito.spy;
 import static org.mockito.Mockito.when;
 
-@Disabled("pending some way to test under Junit5. #554 might have some hope.")
+
+// /!\ WARNING: If you take off this Spf4jTestLogJUnitRunner
+// (e.g. to run as junit5 instead of vintage-junit4), it will
+// initially look like tests pass but the annotations won't work.
+// Make sure you can still get the assertions to *fail* when changed,
+// and that failures still show all the logs specified in CollectLogs
+// annotations.
+//
+// See: https://github.com/zolyfarkas/spf4j/issues/50
+@RunWith(Spf4jTestLogJUnitRunner.class)
 public class TestGameRunner {
 
     /**
      * Passed around to GameRunners to simulate a running game process.
      */
     private Process gameProcess;
+    private TestLoggers testLog;
 
-    @BeforeEach
+    @Before
     public void setup() throws Exception {
-        final LoggerFactory loggerFactory = mock(LoggerFactory.class);
-        mock(Thread.class);
-
-        Logger logger = mock(Logger.class);
-        when(loggerFactory.getLogger(any(Class.class))).thenReturn(logger);
+        testLog = TestLoggers.sys();
 
         // Create fake game process to give to GameRunner
         gameProcess = mock(Process.class);
-        resetGameProcess();
-    }
-
-    /**
-     * Resets the simulated game process to "successful" behaviour for use in multiple tests.
-     *
-     * @throws Exception
-     */
-    private void resetGameProcess() throws Exception {
         when(gameProcess.getInputStream()).thenReturn(new ByteArrayInputStream(new byte[0]));
-        when(gameProcess.waitFor()).thenReturn(0);
     }
 
     @Test
+    // CollectLogs(TRACE) isn't necessary for this test's assertions.
+    // Logs are shown on test failure, but the default doesn't include TRACE.
+    // Because TRACE logs are relevant to this test we use this to include them in the output.
```

Comment:
So TRACE logs "isn't necessary for this test's assertions" but "are relevant to this test"? How does this solve 🤔

Reply:
`CollectLogs` isn't necessary for the test's assertions. The expectation sets up to match against `Level.TRACE` either way.

The "include them in the output" refers to the _visible_ output to the befuddled programmer on test failure.